### PR TITLE
fix: tx flow error handling

### PR DESCRIPTION
**Pull request by kyranjamie**
> [Download the latest build](https://github.com/blockstack/stacks-wallet/actions/runs/488039099)

### CODE REVIEW

File: app/modals/transaction/transaction-modal.tsx [hunk 1]
```diff
@@ -111,29 +113,39 @@ export const TransactionModal: FC<TxModalProps> = ({ balance, address }) => {
...
         ...options,
         publicKey: publicKey.toString('hex'),
       });
+    },
+    [publicKey]
+  );
+
+  const signLedgerTransaction = useCallback(
+    async (unsignedTx: StacksTransaction) => {
+      if (!blockstackApp) throw new Error('``blockstackApp` is not defined');
```
Comment (aulneau) [inline app/modals/transaction/transaction-modal.tsx:148]: `hiroApp`??

Comment (aulneau) [reply app/modals/transaction/transaction-modal.tsx:148]: stxApp??

Comment (aulneau) [reply app/modals/transaction/transaction-modal.tsx:148]: STACKS_APP???

**Review by aulneau — APPROVED** [review]

File: app/modals/transaction/transaction-modal.tsx (hunk 1 above)
Comment (kyranjamie) [reply app/modals/transaction/transaction-modal.tsx:148]: This code definitely needs to be refactored 😅

**Pull request merged**